### PR TITLE
add config option to limit external command completions

### PR DESCRIPTION
**Pull request by merelymyself**
# Description

For increased configurability. No related issue, just something I like.

# Tests

Make sure you've done the following:

- [ ] Add tests that cover your changes, either in the command examples, the crate/tests folder, or in the /tests folder.
- [ ] Try to think about corner cases and various ways how your changes could break. Cover them with tests.
- [ ] If adding tests is not possible, please document in the PR body a minimal example with steps on how to reproduce so one can verify your change works.

I'm not sure how to test this change, so... just use it?

Make sure you've run and fixed any issues with these commands:

- [ ] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [ ] `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [ ] `cargo test --workspace --features=extra` to check that all the tests pass

### CODE REVIEW

File: docs/sample_config/default_config.nu [hunk 1]
```diff
@@ -256,6 +256,7 @@ let-env config = {
   cd_with_abbreviations: false # set to true to allow you to do things like cd s/o/f and nushell expand it to cd some/other/folder
   case_sensitive_completions: false # set to true to enable case-sensitive completions
   enable_external_completion: true # set to false to prevent nushell looking into $env.PATH to find more suggestions, `false` recommended for WSL users as this look up my be very slow
+  max_external_command_completions: 100
```
Comment (WindSoilder) [inline docs/sample_config/default_config.nu:259]: Maybe we can have a shorter name?  Like `max_external_cmd_completions`

And I think it's better to have a comment for it, like other items.

Comment (merelymyself) [reply docs/sample_config/default_config.nu]: Good call, changed it to `max_external_completions` in line with `enable_external_completion`

**Comment by fdncred** [comment]
Thanks for this, but I didn't know what this was until reading the code. I think the name should be `max_external_completion_results` since it's meant to limit the number of completion results. 

Not to conflict with this one but it makes me wonder if we should have a limit on all completion results?

**Comment by fdncred** [comment]
thanks for the updates. let's try it out.

**Comment by fdncred** [comment]
Darn it! what am i doing wrong? I set it to 10 but i'm getting 30, which i presume is all of them. I'm doing `> ca<tab>`.

**Comment by fdncred** [comment]
@merelymyself any word on what I mentioned above?

**Comment by merelymyself** [comment]
Just did a quick test on my end. The number seems to hold correct for me, and custom commands for me are not counted as 'externals'. i.e. if I set the config value to 1 and have a similarly-named custom command to an actual external command, both show up without descriptions. Is that what's happening to you?

**Comment by fdncred** [comment]
> custom commands for me are not counted as 'externals'

I think this was a new PR to show custom commands as externals.

I think it's working as expected on Mac. I'm going to try on Windows a little later because that's where I was seeing the issue yesterday. Thanks!

**Comment by fdncred** [comment]
ok, i get 10 or 11 on windows. close enough for me. thanks for talking it through with me.

**Pull request merged**